metricflow/query/group_by_item/filter_spec_resolution/filter_spec_resolver.py

```diff
@@ -91,66 +94,111 @@ def __init__(self, manifest_lookup: SemanticManifestLookup) -> None:  # noqa: D
         self._path_from_start_node_tracker = DagTraversalPathTracker()
 
     @staticmethod
-    def _map_filter_parameter_set_to_pattern(
+    def _dedupe_filter_call_parameter_sets(
+        filter_call_parameter_sets_sequence: Sequence[FilterCallParameterSets],
+    ) -> FilterCallParameterSets:
+        dimension_call_parameter_sets: List[DimensionCallParameterSet] = []
+        time_dimension_call_parameter_sets: List[TimeDimensionCallParameterSet] = []
+        entity_call_parameter_sets: List[EntityCallParameterSet] = []
+
+        # FilterCallParameterSets needs an update.
+        for filter_call_parameter_sets in filter_call_parameter_sets_sequence:
+            for dimension_call_parameter_set in filter_call_parameter_sets.dimension_call_parameter_sets:
+                if dimension_call_parameter_set not in dimension_call_parameter_sets:
+                    dimension_call_parameter_sets.append(dimension_call_parameter_set)
+            for time_dimension_call_parameter_set in filter_call_parameter_sets.time_dimension_call_parameter_sets:
+                if time_dimension_call_parameter_set not in time_dimension_call_parameter_sets:
+                    time_dimension_call_parameter_sets.append(time_dimension_call_parameter_set)
+            for entity_call_parameter_set in filter_call_parameter_sets.entity_call_parameter_sets:
+                if entity_call_parameter_set not in entity_call_parameter_sets:
+                    entity_call_parameter_sets.append(entity_call_parameter_set)
```

Comment:
I changed to this to track down a testing consistency issue, so partially changed it back. This will get rewritten with the change to `FilterCallParameterSets`.